src/user_override.cc

```diff
@@ -44,38 +80,66 @@ static DDRes setresuid_wrapper(uid_t ruid, uid_t euid, uid_t suid) {
   // (ie. parent process) needs to be able to read ddprof /proc/<pid>/fd/*,
   // that's why we set dumpable attribute back to its intial value at each
   // effective user id change.
-  prctl(PR_SET_DUMPABLE, dumpable);
+  DumpableRestorer dumpable_restorer;
+  if (gid != static_cast<uid_t>(-1)) {
+    DDRES_CHECK_INT(setresgid(gid, gid, -1), DD_WHAT_USERID,
+                    "Unable to set gid %d (%s)", gid, strerror(errno));
+  }
+  if (uid != static_cast<uid_t>(-1)) {
+    DDRES_CHECK_INT(setresuid(uid, uid, -1), DD_WHAT_USERID,
+                    "Unable to set uid %d (%s)", uid, strerror(errno));
+  }
+
   return {};
 }
 
```

Comment:
Wow, thanks for spending some amount of time figuring this out ! That is a lot of operations. Interesting that the group is necessary.